marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/OutputEndpointImpl.java

```diff
@@ -59,7 +62,59 @@ private BulkOutputCallerImpl(OutputEndpointImpl endpoint) {
         @Override
         public void awaitCompletion() {
             logger.trace("output endpoint running endpoint={} work={}", getEndpointPath(), getWorkUnit());
+            setPhase(WorkPhase.RUNNING);
+            SessionState session = allowsSession() ? getEndpoint().getCaller().newSessionState() : null;
 
+            calling: while (true) {
+                Stream<InputStream> output = null;
+                try {
+                    logger.trace("output endpoint={} count={} state={}",
+                            getEndpointPath(), getCallCount(), getEndpointState());
+
+                    output = getEndpoint().getCaller().call(
+                            getEndpoint().getClient(), getEndpointState(), session, getWorkUnit()
+                    );
+                    incrementCallCount();
+                } catch(Throwable throwable) {
+                    throw new RuntimeException("error while calling "+getEndpoint().getEndpointPath(), throwable);
+                }
+
+                if (allowsEndpointState()) {
+                    if(output!= null && output.findFirst()!= null && output.findFirst().isPresent()) {
+                        InputStream[] result = output.toArray(size -> new InputStream[size]);
+                        setEndpointState(result[0]);
+                        for(int i=1; i<result.length; i++) {
+                            forEachOutput((Consumer<InputStream>) result[i]);
```

Comment:
forEachOutput() is a setter for a Consumer function.  The loop should call the Consumer function's accept() method.